### PR TITLE
Playerclass/executable showing class info, class specific moves and class item restrictions

### PR DESCRIPTION
**Pull request by simrenk**
Resolves issue #669 
In this demo, we showcase examples of classes and their moves.
After a mock battle occurs, we show how class item restrictions might affect the player's ability to harness their loot.

### CODE REVIEW

File: src/playerclass/examples/demo-functions.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/* See demo-functions.h */
+void class_show(class_t* c)
+{
+    if (c == NULL)
+    {
+        fprintf(stderr, "Cannot show information of a NULL class.\n");
+        exit(1);
+    }
+        fprintf(stdout, "Class info:\n");
+        frpintf(stdout, "    Name: %s\n", c->name);
```
Comment (PaulTreitel) [inline src/playerclass/examples/demo-functions.c:16]: Spelling mistake. Should be "fprintf" instead of "frpintf"

File: src/playerclass/examples/demo-functions.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    move_list_t* tmp;
+
+    DL_FOREACH(m, tmp)
+    {   
+        class_show(tmp->c);
+        fprintf(stdout, "    Special Skill: %s\n", tmp->spell);
+        fprintf(stdout, "    Skill Damage: %d\n", tmp->damage);
+        frpintf(stdout, "    Skill id: %d\n", tmp->id);
```
Comment (PaulTreitel) [inline src/playerclass/examples/demo-functions.c:37]: Same spelling mistake here

File: src/playerclass/examples/CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,21 @@
...
+            class-list.c  
+            demo-functions.c
             )
 
 target_include_directories(playerclass_examples PRIVATE src/)
 
+set(PLAYERCLASS_EXAMPLES pc_demo)
+
+add_executable(pc_demo
+               pc_demo.c)
```
Comment (PaulTreitel) [inline src/playerclass/examples/CMakeLists.txt:11]: I figured out why it wasn't building. The commands here build it like it's a library, not an executable. Drop the `add_library` and `target_include_directories` commands and add the two `.c` files in them into the `add_executable` command.

Comment (ethanwtai) [reply src/playerclass/examples/CMakeLists.txt:11]: That seems to have resolved it thanks! I'm keeping class-list.c in the library since I know battle systems uses the functions inside which seems to not break the build.

File: src/playerclass/examples/demo-functions.c [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+{
+    if (c == NULL)
+    {
+        fprintf(stderr, "Cannot show information of a NULL class.\n");
+        exit(1);
+    }
+        fprintf(stdout, "Class info:\n");
+        fprintf(stdout, "    Name: %s\n", c->name);
+        fprintf(stdout, "    Short Description: %s\n", c->shortdesc);
+        fprintf(stdout, "    Long Description: %s\n", c->longdesc);
```
Comment (dwahme) [inline src/playerclass/examples/demo-functions.c:18]: Why are we using `fprintf` instead of `printf`?

File: src/playerclass/examples/demo-functions.c [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    move_list_t* tmp;
+
+    DL_FOREACH(m, tmp)
+    {   
+        class_show(tmp->c);
+        fprintf(stdout, "    Special Skill: %s\n", tmp->spell);
+        fprintf(stdout, "    Skill Damage: %d\n", tmp->damage);
+        fprintf(stdout, "    Skill id: %d\n", tmp->id);
```
Comment (dwahme) [inline src/playerclass/examples/demo-functions.c:37]: Why are we using `fprintf` instead of `printf`?

File: src/playerclass/examples/pc_demo.c [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    list_class_show(classlist);
+
+    fprintf(stdout, "You have chosen: \n");
+    class_show(classlist->c);
+
+    fprintf(stdout, "Proceeding with battle.\n");
+    
+    fprintf(stdout, "***Battle occurs here***\n");
+
+    fprintf(stdout,"Battle has ended. You have won.\n");
```
Comment (dwahme) [inline src/playerclass/examples/pc_demo.c:35]: Why are we using `fprintf` instead of `printf`?

File: src/playerclass/examples/pc_demo.c [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    
+    item_t* wand = item_new("wand", " ", " ");
+    item_t* sword = item_new("sword", " ", " ");
+    item_t* guitar = item_new("guitar", " "," ");
+
+    set_item_restriction(wand, classlist->c);
+    set_item_restriction(sword, classlist->c);
+    
+    fprintf(stdout, "Your enemy has dropped the following items: %s, %s, %s. You attempt to pick them up with mixed results.\n",
+	    wand->item_id, sword->item_id, guitar->item_id);
```
Comment (dwahme) [inline src/playerclass/examples/pc_demo.c:45]: Why are we using `fprintf` instead of `printf`?

File: src/playerclass/examples/pc_demo.c [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    bool not_allowed_guitar = get_class_restriction(guitar, classlist->c);
+
+    fprintf(stdout, "You are %s to use the  %s, %s to use the %s and %s to use the %s because you are a %s \n",
+	    not_allowed_wand? "not able":"able",
+	    wand->item_id,
+	    not_allowed_sword? "not able":"able",
+	    sword->item_id,
+	    not_allowed_guitar? "not_able":"able",
+	    guitar->item_id,
+	    classlist->c->name);
```
Comment (dwahme) [inline src/playerclass/examples/pc_demo.c:58]: Why are we using `fprintf` instead of `printf`?

File: include/game-state/player.h [hunk 9]
```diff
@@ -6,6 +6,7 @@
 #ifndef _PLAYER_H
 #define _PLAYER_H
 
+#include "playerclass/class.h"
```
Comment (dwahme) [inline include/game-state/player.h:9]: Let's remove unnecessary changes to other files, particularly ones outside our modules

File: src/game-state/src/player.c [hunk 10]
```diff
@@ -22,7 +22,7 @@ player_t* player_new(char* player_id, int health)
     plyr = malloc(sizeof(player_t));
     memset(plyr, 0, sizeof(player_t));
     plyr->player_id = malloc(MAX_ID_LEN);
-
+
```
Comment (dwahme) [inline src/game-state/src/player.c:25]: Let's remove unnecessary changes to other files, particularly ones outside our modules

File: src/game-state/src/player.c [hunk 11]
```diff
@@ -148,4 +148,4 @@ int assign_stats_player(player_t *plyr, stats_hash_t *sh)
 {
     printf("assign_stats_player: function not yet implemented\n");
     return 0; // still needs to be implemented
-}
\ No newline at end of file
+}
```
Comment (dwahme) [inline src/game-state/src/player.c:151]: Let's remove unnecessary changes to other files, particularly ones outside our modules (remove the newline after the brace I think here)

File: include/game-state/player.h (hunk 9 above)
Comment (simrenk) [reply include/game-state/player.h:9]: whoops sorry - didn't mean to add this

**Comment by dwahme** [comment]
Issue Score: ✔️

Comments:
Great job! Here are a few things you may want to improve in future issues:

The summary for this issue is too terse. A good rule of thumb is that you should be writing at least one full paragraph in your summary (see https://uchicago-cs.github.io/cmsc22000/project/policies/#a-descriptive-summary for examples). If this issue was generated from a Scrum board card, please remember that the summary you write in the Sprint meeting is meant to be a quick draft; it needs to be fleshed out when you convert the card to an issue.

When an issue is done, you should include a final comment summarizing what was done, and why the issue is being closed.
